src/ui_components/list.js

```diff
@@ -51,6 +51,15 @@ class List extends React.Component {
         var listItems = [];
 
         for (let gifter in gifters) {
+
+            // Adds a distanceToUser attribute to the array, used for list sorting
+            gifters[gifter]["distanceToUser"] = this.calculateDistanceTo(gifters[gifter].coords)
+
+            // Sort the list by distance, ascending
```

Comment:
Here you are doing the sorting within the iteration, I think that's what's causing the problem when it's rendered the first time:
![image](https://user-images.githubusercontent.com/294776/40918308-33ee964e-6806-11e8-81e5-d0744a7c5615.png)

Try running through a whole nother loop beforehand to do the sorting?